driver-scala/src/main/scala/org/mongodb/scala/model/Windows.scala

```diff
@@ -58,6 +59,24 @@ import org.bson.types.Decimal128
 @Beta
 object Windows {
 
+  /**
+   * Creates a window from a document in situations when there is no builder method that better satisfies your needs.
+   * This method cannot be used to validate the document syntax.
+   *
+   * <i>Example</i><br>
+   * The following code creates two functionally identical windows, though they may not be equal.
+   * {{{
+   *  val pastWeek1: Window = Windows.timeRange(-1, MongoTimeUnit.WEEK, Windows.Bound.CURRENT)
+   *  val pastWeek2: Window = Windows.of(
+   *      Document("range" -> BsonArray(-1, "current"),
+   *          "unit" -> BsonString("week")))
+   * }}}
+   *
+   * @param window A document representing the required window.
+   * @return The constructed window.
+   */
+  def of(window: Document): Window = JWindows.of(window)
```

Comment:
There is only one `Windows.of` method in the Scala API because as far as I understand from the docs saying [_"The underlying implementations of the scala Document use the type safe BsonDocument class"_](), there is no direct counterpart of `org.bson.Document` in the Scala API, and there is no split as opposed to `org.bson.BsonDocument` and `org.bson.Document` as in the Java API.

Is my understanding correct and we do not need the overload here?

Reply:
This comment is not relevant anymore.